src/tike/ptycho/solvers/lstsq.py

```diff
@@ -137,35 +137,36 @@ def lstsq_grad(
             exitwave_options=exitwave_options,
             op=op,
             recover_psi=object_options is not None,
-            recover_probe=probe_options is not None,
+            recover_probe=probe_options.recover_probe,
             recover_positions=position_options is not None,
         )))
 
         if object_options is not None:
             object_upd_sum = comm.Allreduce(object_upd_sum)
 
-        if probe_options is not None:
+        if probe_options.recover_probe :
             m_probe_update = comm.pool.bcast(
                 [comm.Allreduce_mean(
                     m_probe_update,
                     axis=-5,
                 )])
 
-        (
-            beigen_probe,
-            beigen_weights,
-        ) = _update_nearplane(
-            comm,
-            diff,
-            probe_update,
-            m_probe_update,
-            probe,
-            beigen_probe,
-            beigen_weights,
-            patches,
-            batches,
-            batch_index=batch_index,
-        )
+        if probe_options.recover_probe :
```

Comment:
Use the indentation block above instead of checking the same condition twice in a row.

Reply:
fixed I think